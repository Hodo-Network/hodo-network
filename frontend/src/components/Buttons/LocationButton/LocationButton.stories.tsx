import { Meta, Story } from '@storybook/react';

import { IPureLocationButtonProps, PureLocationButton } from './LocationButton';

export default {
  title: 'Components/Buttons/LocationButton',
  component: PureLocationButton,
} as Meta;

const Template: Story<IPureLocationButtonProps> = (args) => (
  <PureLocationButton {...args} />
);

export const Default = Template.bind({});
Default.args = {};

export const LocationSet = Template.bind({});
LocationSet.args = {
  location: '0, 0',
};
