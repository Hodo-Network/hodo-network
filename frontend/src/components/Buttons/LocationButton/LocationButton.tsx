import Button from "../../../base/Button";

export interface IPureLocationButtonProps {
  label?: string;
  location?: string;
  onClick?: () => void;
}

export const PureLocationButton = ({ label = "Get Location", location, onClick }: IPureLocationButtonProps) => {
  if (location) {
    return (
      <span>
        {location}
      </span>
    );
  } else {
    return (
      <Button color="primary" onClick={onClick}>
        {label}
      </Button>
    );
  }
};
