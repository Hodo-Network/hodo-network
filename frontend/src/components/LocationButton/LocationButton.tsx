import { useState } from "react";
import useGeolocation from "@rooks/use-geolocation";
import { Button } from "../Button";

export const LocationButton = () => {
  const [when, setWhen] = useState(false);

  const geoObj = useGeolocation({
    when,
  });

  const onClick = () => {
    setWhen(true);
  };

  return (
    <>
      <Button primary={false} size='large' onClick={onClick}>
        Get Location
      </Button>
      <div className='mt-2 text-sm text-gray-500'>
        {geoObj && JSON.stringify(geoObj)}
      </div>
    </>
  );
};
