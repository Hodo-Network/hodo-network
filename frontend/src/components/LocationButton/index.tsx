import { useState } from "react";
import useGeolocation from "@rooks/use-geolocation";

export default function LocationButton() {
  const [when, setWhen] = useState(false);

  const geoObj = useGeolocation({
    when,
  });

  return (
    <>
      <button
        onClick={() => {
          setWhen(true);
        }}
        className='btn btn--large w-full'>
        Get Location
      </button>
      <div className='mt-2 text-sm text-gray-500'>
        {geoObj && JSON.stringify(geoObj)}
      </div>
    </>
  );
}
