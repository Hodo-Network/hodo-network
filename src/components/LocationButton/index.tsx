import { useState } from "react";
import useGeolocation from "@rooks/use-geolocation";

export default function LocationButton() {
  const [when, setWhen] = useState(false);

  const geoObj = useGeolocation({
    when,
  });

  return (
    <>
      <button
        onClick={() => {
          setWhen(true);
        }}
        className='inline-flex items-center justify-center px-5 py-3 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 dark:bg-indigo-500 dark:hover:bg-indigo-600'>
        Get Location
      </button>
      <div className='mt-2 text-sm text-gray-500'>
        {geoObj && JSON.stringify(geoObj)}
      </div>
    </>
  );
}
