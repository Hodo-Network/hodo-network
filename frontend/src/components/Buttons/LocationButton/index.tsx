import { useState } from "react";
import useGeolocation from "@rooks/use-geolocation";
import { PureLocationButton } from './LocationButton';

export const LocationButton = () => {
  const [when, setWhen] = useState(false);

  const geoObj = useGeolocation({
    when,
  });

  const onClick = () => {
    setWhen(true);
  };

  {/* JSON.stringify(geoObj) */ }
  return <PureLocationButton location={`${geoObj?.lat}, ${geoObj?.lng}`} onClick={onClick} />;
};

export default LocationButton;
