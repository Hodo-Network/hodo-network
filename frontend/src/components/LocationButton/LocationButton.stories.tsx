import { Story, Meta } from "@storybook/react";

import { LocationButton } from "./LocationButton";

export default {
  title: "Components/LocationButton",
  component: LocationButton,
} as Meta;

const Template: Story = (args) => <LocationButton {...args} />;

export const Default = Template.bind({});
Default.args = {};
