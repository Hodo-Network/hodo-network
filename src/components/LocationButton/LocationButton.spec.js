import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import LocationButton from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<LocationButton />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<LocationButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
