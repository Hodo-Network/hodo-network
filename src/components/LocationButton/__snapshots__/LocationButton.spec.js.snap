// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <button
    className="inline-flex items-center justify-center px-5 py-3 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 dark:bg-indigo-500 dark:hover:bg-indigo-600"
    onClick={[Function]}
  >
    Get Location
  </button>,
  <div
    className="mt-2 text-sm text-gray-500"
  />,
]
`;
